src/microbe_stage/Microbe.Contact.cs

```diff
@@ -964,76 +1129,93 @@ private void HandleEngulfing(float delta)
             }
         }
 
-        // Check whether we should not be being engulfed anymore
-        var hostile = hostileEngulfer.Value;
-        if (hostile != null)
-        {
-            // Dead things can't engulf us
-            if (hostile.Dead)
-            {
-                hostileEngulfer.Value = null;
-                IsBeingEngulfed = false;
-            }
-        }
-        else
+        for (int i = engulfedObjects.Count - 1; i >= 0; --i)
         {
-            IsBeingEngulfed = false;
-        }
+            var engulfedObject = engulfedObjects[i];
 
-        previousEngulfMode = State == MicrobeState.Engulf;
-    }
+            var engulfable = engulfedObject.Object.Value;
```

Comment:
Does it make sense if engulfable is null here? Would it be better to just bail out (I see quite a lot of null forgiving operator use in this section)?